src/steps/05-PerformanceRequirements/DOW/ComputeForm.spec.ts

```diff
@@ -0,0 +1,700 @@
+          "option_order": "1"
+        },
+          {
+          "period_unit": "YEAR",
+          "period_unit_count": "1",
+          "period_type": "BASE",
+          "option_order": "2"
+        },
+        {
+          "period_unit": "YEAR",
+          "period_unit_count": "1",
+          "period_type": "BASE",
+          "option_order": "3"
+        }
+      ]
+    ]
+
+  beforeEach(() => {
+    vuetify = new Vuetify();
+    wrapper = mount(ComputeForm, {
+      localVue,
+      vuetify,
+      mocks:{
+        $store:{
+          DescriptionOfWork:{
+            computeObject: computeData,
+          }
+        },
+        
+      },
+      propsData: {
+        computeData: computeData
+      }
+    });
+  });
+
+
+  describe("Initialization....", () => {
+    it("tests that component renders successfully", async () => {
+      expect(wrapper.exists()).toBe(true);
+    });
+  });
+
+  describe("Vue Lifecyle Hook Testing...", ()=>{
+    it("mounted", ()=>{
+      expect(wrapper.vm.$data.formHasBeenTouched).toBe(false);
+    })
+  })
+
+  describe("Method Testing...", ()=>{
+    it ("loadOnEnter() - is to set data.firstTimeHere to true ", async()=>{
+      const spy = await jest.spyOn(wrapper.vm, "loadOnEnter");
+      wrapper.vm.loadOnEnter();
+      expect(wrapper.vm.$data.firstTimeHere).toBe(true);
+    });
+
+    it("@Watch errorBagChange() - sets data.formHasErrors when errors occur ", async()=>{
+      // set the initial data.errorBagValues
+      wrapper.vm.$data.errorBagValues = [false, false];
+      const computeForm = wrapper.findComponent({ref: "computeForm"});
+      computeForm.vm.$data.errorBag = {810:true, 827: false}
+
+      // change the data.errorBagValues value to activate data reactivity
+      await wrapper.setData({
+        errorBagValues: [true, false]
+      })
+      expect(await wrapper.vm.$data.formHasErrors).toBe(true);
+    });
+  
+    it("validate() - ensures validate() has been called", async ()=>{
+      const validateSpy = await jest.spyOn(wrapper.vm, "validate");
+      wrapper.vm.validate();
+      expect(validateSpy).toHaveBeenCalled();
+    })
+
+    it("createPeriodCheckboxItems() - tests that unsorted SNOW data is successfully transformed to " +
+       "expected sorted datasource array for period checkbox items", async () => {
+      const spy = await jest.spyOn(wrapper.vm, "createPeriodCheckboxItems");
+      const _createPeriodCheckboxItems = wrapper.vm.createPeriodCheckboxItems(
+        [ 
+          {
+            "period_unit": "YEAR",
+            "period_unit_count": "1",
+            "period_type": "BASE",
+            "option_order": "2",
+            "sys_id": "period_02"
+          },
+          {
+            "period_unit": "YEAR",
+            "period_unit_count": "1",
+            "period_type": "BASE",
+            "option_order": "1",
+            "sys_id": "period_01"
+
+          },
+        ]
+      );        
+      expect(spy).toHaveBeenCalled();
```

Comment:
Since we're calling the method ourselves, this seems redundant

Reply:
trying to make it 'bulletproof.' Looks like I made it a little too bulletproof.

Reply:
Removed all spy references, as they are are not needed.  Thx!